views/js/runner/helpers/offlineJumpTable.js

```diff
@@ -0,0 +1,420 @@
+
+                return this;
+            },
+
+            /**
+             * Returns the jump table.
+             *
+             * @returns {Array}
+             */
+            getJumpTable: function getJumpTable() {
+                return jumpTable;
+            },
+
+            /**
+             * Returns the last entry of the jump table which represent the current state of the navigation.
+             */
+            getLastJump: function getLastJump() {
+                return jumpTable.length > 0
+                    ? jumpTable[jumpTable.length - 1]
+                    : {};
+            },
+
+            /**
+             * Returns all the item identifiers in order.
+             *
+             * @returns {Array}
+             * @private
+             */
+            _getItems: function _getItems() {
+                return this._getSimplifiedTestMap()
+                    .map(function(row) {
+                        return row.item;
+                    })
+                    .filter(function(value, index, thisArg) {
+                        return thisArg.indexOf(value) === index;
+                    });
+            },
+
+            /**
+             * Returns all the section identifiers in order.
+             *
+             * @returns {Array}
+             * @private
+             */
+            _getSections: function _getSections() {
+                return this._getSimplifiedTestMap()
+                    .map(function(row) {
+                        return row.section;
+                    })
+                    .filter(function(value, index, thisArg) {
+                        return thisArg.indexOf(value) === index;
+                    });
+            },
+
+            /**
+             * Returns all the part identifiers in order.
+             *
+             * @returns {Array}
+             * @private
+             */
+            _getParts: function _getParts() {
+                return this._getSimplifiedTestMap()
+                    .map(function(row) {
+                        return row.part;
+                    })
+                    .filter(function(value, index, thisArg) {
+                        return thisArg.indexOf(value) === index;
+                    });
+            },
+
+            /**
+             * Returns a simplified test map array, which will contain the item, section and part identifiers.
+             *
+             * @returns {Array}
+             * @private
+             */
+            _getSimplifiedTestMap: function _getSimplifiedTestMap() {
+                var simplifiedTestMap = [];
+
+                Object.keys(testMap.parts).forEach(function(partIdentifier) {
+                    var part = testMap.parts[partIdentifier];
+                    Object.keys(part.sections).forEach(function(sectionIdentifier) {
+                        var section = part.sections[sectionIdentifier];
+                        Object.keys(section.items).forEach(function(itemIdentifier) {
```

Comment:
Using `lodash` will allow to:
- have a more readable code
- avoid additional loops
- avoid to create intermediate variables

```javascript
_.forEach(testMap.parts, function(part, partIdentifier) {
    _.forEach(part.sections, function(section, sectionIdentifier) {
        _.forEach(section.items, function(item, itemIdentifier) {
            ...
        });
    });
});
```